core/src/main/java/pt/webdetails/cpf/utils/AbstractCorsUtil.java

```diff
@@ -0,0 +1,120 @@
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. Please refer to
+ * the license for the specific language governing your rights and limitations.
+ */
+
+package pt.webdetails.cpf.utils;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.util.CollectionUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+
+/**
+ * An abstract class to provide a carcass for a CORS implementation in plugins.
+ * The idea is in populating an HTTP response with a CORS headers, allowing access
+ * from a domain specified by a user in an {@link #ORIGIN} header if it is also listed
+ * in an whitelist {@link #getDomainWhitelist()}.
+ *
+ * @author Yahor_Zhuk
+ */
+public abstract class AbstractCorsUtil {
+
+  private static final String TRUE = "true";
+  private static final String ORIGIN = "ORIGIN";
+  private static final String ACCESS_CONTROL_ALLOW_ORIGIN = "Access-Control-Allow-Origin";
+  private static final String ACCESS_CONTROL_ALLOW_CREDENTIALS = "Access-Control-Allow-Credentials";
+  private static final String COMMA = ",";
+
+  /**
+   * The value is assumed to be retrieved from a plugin configuration file.
+   * In the default implementation we use a property with a name <i>allow-cross-domain-resources</i>,
+   * so it is a good idea to use the same name in your own implementation.
+   *
+   * @return {@link java.lang.Boolean#TRUE} if cross-domain calls are allowed
+   */
+  protected abstract boolean isCorsAllowed();
+
+  /**
+   * The values are assumed to be retrieved from a plugin configuration file.
+   * In the default implementation we use a property with a name <i>cross-domain-resources-whitelist</i>,
+   * so it is a good idea to use the same name in your own implementation
+   *
+   * @return collection of allowed domains
+   */
+  protected abstract Collection<String> getDomainWhitelist();
+
+  /**
+   * Sets CORS headers to the response if the domain provided by a user is in a whitelist
+   *
+   * @param request  a request, nothing special
+   * @param response a response, nothing special
+   */
+  public void setCorsHeaders( HttpServletRequest request, HttpServletResponse response ) {
+    if ( isCorsAllowed() ) {
+      final String origin = request.getHeader( ORIGIN );
+      if ( isDomainAllowed( origin ) ) {
+        response.setHeader( ACCESS_CONTROL_ALLOW_ORIGIN, origin );
+        response.setHeader( ACCESS_CONTROL_ALLOW_CREDENTIALS, TRUE );
+      }
+    }
+  }
+
+  /**
+   * An utility method used in the default implementations.
+   * Is intentionally left here in order not to implement it in each plugin.
+   *
+   * @param csvString a comma separated string
+   * @return a collection of strings, never a null
+   */
+  protected Collection<String> parseCsvString( final String csvString ) {
```

Comment:
I get that, but this should be in a StringUtils or CSVUtils or whatever. But ok, I guess it's something that can be moved later.